api/app/signals/apps/api/v1/views/csv.py

```diff
@@ -0,0 +1,68 @@
+import glob
+import os
+
+from django.conf import settings
+from django.http import FileResponse
+from django.utils import timezone
+from rest_framework import renderers, status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from signals.apps.api.generics.permissions import SIAPermissions, SIAReportPermissions
+from signals.auth.backend import JWTAuthBackend
+
+
+class PassthroughRenderer(renderers.BaseRenderer):
+    """
+        Return data as-is. View should supply a Response.
+    """
+    media_type = ''
+    format = ''
+
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        return data
+
+
+class PrivateCsvViewSet(viewsets.ViewSet):
+    """
+    V1 private ViewSet to retrieve generated csv files
+    https://stackoverflow.com/a/51936269
+    """
+
+    authentication_classes = (JWTAuthBackend, )
+    permission_classes = (SIAPermissions & SIAReportPermissions, )
+
+    @action(methods=['get'], detail=True, renderer_classes=(PassthroughRenderer,))
+    def download(self, *args, **kwargs):
```

Comment:
Why a specific action? Why not just overwrite the `get` to do the download?

Reply:
check